### PR TITLE
Add `ecosystem.json` to include project in Qiskit ecosystem

### PR DESCRIPTION
**Pull request by ElePT**
### Summary
Adds an `ecosystem.json` file to pass necessary checks to add project to Qiskit ecosystem.

### Details and comments
- Fixes build shown in https://github.com/qiskit-community/ecosystem/issues/484
- See ecosystem submission guidelines here: https://github.com/qiskit-community/ecosystem/blob/main/docs/project_overview.md#adding-project-to-the-ecosystem
- I am following the default template as suggested, I hope this is enough.

### CODE REVIEW

**Review by Cryoris — COMMENTED** [review]
I would expect we need to make some changes from the example file, as we use different tools and already have some workflows in place 🙂 But I'm also not sure if this is correct -- however opening an issue and letting the ecosystem's CI run would probably resolve it 😄

**Comment by woodsp-ibm** [comment]
I would have to ask whether this is really needed - it was not done for any of the apps. I guess my concern is over running duplicate? CI over the tests? I get that if you are submitting a new project having some CI run is good so ecosystem provisioing something is helpful - where is this run under the ecosystem's repo github actions. But in this case we have a comprehensive CI that runs already here - and flags in the readme, via the ribbon, whether tests are passing etc which I guess ends up as as outcome of whatever ecosystem runs. Would it not be better to just link this in someway assuming this is doable in the ecosystem. If it was setup for this it could hooked up for the apps too so their status was visible in the ecosystem project cards.

File: ecosystem.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    ],
+    "extra_dependencies": [
+        "pytest"
+    ],
+    "language": {
+        "name": "python",
+        "versions": ["3.9"]
+    },
+    "tests_command": [
+        "pytest"
```
[the remark this answers is not in the document]
Comment (mickahell) [reply ecosystem.json:14]: Just keep in mind in Ecosystem we launch it inside a tox env, so keep it as simple as possible :)

File: ecosystem.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    ],
+    "styles_check_command": [
+        "pylint -rn src tests"
```
[the remark this answers is not in the document]
Comment (mickahell) [reply ecosystem.json:17]: Just keep in mind in Ecosystem we launch it inside a tox env, so keep it as simple and basic as possible :)

File: ecosystem.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+    "dependencies_files": [
+        "requirements.txt",
+        "requirements-dev.txt"
+    ],
+    "extra_dependencies": [
+        "pytest"
```
[the remark this answers is not in the document]
Comment (mickahell) [reply ecosystem.json:7]: Yes you can remove the whole section :)

**Comment by mickahell** [comment]
> I would have to ask whether this is really needed - it was not done for any of the apps. I guess my concern is over running duplicate? CI over the tests? I get that if you are submitting a new project having some CI run is good so ecosystem provisioing something is helpful - where is this run under the ecosystem's repo github actions. But in this case we have a comprehensive CI that runs already here - and flags in the readme, via the ribbon, whether tests are passing etc which I guess ends up as as outcome of whatever ecosystem runs. Would it not be better to just link this in someway assuming this is doable in the ecosystem. If it was setup for this it could hooked up for the apps too so their status was visible in the ecosystem project cards.

Hello @woodsp-ibm :)

I'm Michael one of the creator of the Ecosystem and I'm trying to continue contributing to.
The file is mainly necessary to take into account your `requirements-dev.txt`, by default we only take the classical one. The state of the `ecosystem.json` file @ElePT pushed is mainly the template one and the one I'm suggesting to be as clause as possible to any project who need one ([very basic template repo](https://github.com/mickahell/qiskit-ecosystem_template/blob/main/ecosystem.json)).

If you want to test before merging you can set the branch of this PR as default and submitting it in an Ecosystem fork (I can do it for you if you want)

**Comment by woodsp-ibm** [comment]
@mickahell If we need to have a json file its fine by me. My comment was more around the fact that the repo heres has its own CI and does not need anything run elsewhere - and most likely what is here is way more comprehensive. Couple this with the fact that any number of other repos are part of Ecosystem (qiskit-aer, runtime, provider, experiments, nature, etc) that do not have an ecosystem.json, so however these were moved from qiskit org and became ecosystem projects I am not quite sure why this is being treated different. Not that I mind that it is as such, more I was questioning things since the changes noted above all seem to relate to CI being run for this - in the ecosystem repo I think if I understand how things are. How that's triggered and when it runs I have no idea. And since we have CI here with info about tests passing etc I guess I was wondering if that's all that was wanted to be shown on the ecosystem card (though I note none of the other repo's above have that either) is there no way to leverage this whereby we could include some link or other in the json for status rather than having to use resources to runs tests just to do that when the whole CI is already run here.

**Comment by mickahell** [comment]
hmm ok, I think it's just a misunderstanding.

Project in Ecosystem are just to present on the page https://qiskit.org/ecosystem for the public. There it has project from IBM, community and partner. Adding a project there doesn't mean your project goes to community project at all. The default tier is community but we'll allow to people to set another tier in the next release.

The process to check : the Ecosystem gonna scan your project if you don't have `ecosystem.json` he gonna execute its defaults commands and generate a tox file (same if you don't have setup.py). By default we only install the basic `requirements.txt` that why you need an `ecosystem.json` because for your tests you need to install your `requirements-dev.txt`.
This process is not for you to check if your project work, it's for people to see with which version of Qiskit your project can works. And also it's a minimum wage of quality.

With every project having their own CI and diff way to do so, we set ours own and we created the `ecosystem.json` to allow people to _translate_ their CI into ours. At the beginning the goal was to only accept projects who pass our quality checks, that's not true today as we help a lot of people to make their project pass and some checks have been disabled.

And about which qiskit project is treat and how, I don't know. I think the main qiskit project are adding by hand with `test off`, but I'm not sure. For that maybe @frankharkins could help us.

I hope, what I wrote is clear enough and  I answer your though @woodsp-ibm :)

File: ecosystem.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (mickahell) [reply ecosystem.json:7]: Just check it, what do you mean ?
I even don't see pytest in your 2 files

File: ecosystem.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mickahell) [reply ecosystem.json:14]: You'll need to add :
```json
    "debian_dependencies": [
        "make"
    ],
```

The feature is actually in review process

File: ecosystem.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (mickahell) [reply ecosystem.json:17]: You'll need to add :
```json
    "debian_dependencies": [
        "make"
    ],
```

The feature is actually in review process

**Comment by frankharkins** [comment]
Sorry I missed this. We'll be dropping the testing feature of the Ecosystem soon, so this PR is no longer needed.

**Comment by woodsp-ibm** [comment]
@frankharkins Thanks - but what we we really want/need the qiskit-algorithms listed in the ecosystem, that was the real goal behind this PR. qiskit-nature etc were all added somehow so they are listed there (like qiskit-experiments, qiskit-aer etc none of these have this file), but qiskit-algorithms does not show up. Maybe someone manually added the apps etc. so that this is yet to be done for this project.

**Comment by frankharkins** [comment]
Sorry, I should have clarified; I've added the project to the Ecosystem in https://github.com/qiskit-community/ecosystem/pull/485, and it should appear on https://qiskit.org/ecosystem when that's next rebuilt (I believe nightly).

Let me know if you need anything else.

**Comment by woodsp-ibm** [comment]
> ...I've added the project to the Ecosystem...

Perfect, thanks.

**Comment by woodsp-ibm** [comment]
`qiskit-algorithms` is now listed as a project in [Qiskit Ecosystem](https://qiskit.org/ecosystem) so I am closing this PR as it's no longer needed as noted above.

**Pull request closed without merging**